lib/components/viewers/nearby/stop.tsx

```diff
@@ -0,0 +1,240 @@
+  setViewedStop: (stop: any, nearby: string) => void
+  showOperatorLogo: boolean
+  stopData: StopData
+  transitOperators?: TransitOperator[]
+  zoomToPlace: (map: any, stopData: any) => void
+}
+
+const Operator = ({ operator }: { operator: TransitOperator | undefined }) => {
+  const intl = useIntl()
+  return operator ? (
+    <OperatorLogo
+      alt={intl.formatMessage(
+        {
+          id: 'components.StopViewer.operatorLogoAriaLabel'
+        },
+        {
+          operatorName: operator.name
+        }
+      )}
+      operator={operator}
+    />
+  ) : (
+    <MapPin />
+  )
+}
+
+const Stop = ({
+  homeTimezone,
+  setHoveredStop,
+  setLocation,
+  setViewedStop,
+  stopData,
+  transitOperators,
+  zoomToPlace
+}: Props): JSX.Element => {
+  const map = useMap()
+
+  const agencies = stopData.stoptimesForPatterns?.reduce<Set<string>>(
+    // @ts-expect-error The agency type is not yet compatible with OTP2
+    (prev, cur) => prev.add(cur.pattern.route.agency.gtfsId),
+    new Set()
+  )
+
+  // TODO: We need to bring back the day break-up we had with the old stop viewer
+  const patternRows = stopData.stoptimesForPatterns
+    ?.reduce<PatternStopTime[]>((acc, cur) => {
+      const dupe = acc.findIndex(
+        (p) => p.pattern.headsign === cur.pattern.headsign
+      )
+      if (dupe === -1) {
+        acc.push(cur)
+      } else {
+        // TODO: is there a method that already does this?
+        const filteredNewStopTimes = cur.stoptimes.filter(
+          (stoptime: StopTime) =>
+            !acc[dupe].stoptimes.find(
+              (s: StopTime) => fullTimestamp(stoptime) === fullTimestamp(s)
+            )
+        )
+        acc[dupe].stoptimes = [...acc[dupe].stoptimes, ...filteredNewStopTimes]
+      }
+      return acc
+    }, [])
+    .map((st: any, index: number) => {
+      const sortedStopTimes = st.stoptimes.sort(
+        (a: StopTime, b: StopTime) => fullTimestamp(a) - fullTimestamp(b)
+      )
+      return (
+        <PatternRow
+          homeTimezone={homeTimezone}
+          key={index}
+          pattern={st.pattern}
+          roundedTop={false}
+          route={st.pattern.route}
+          stopTimes={sortedStopTimes}
+        />
+      )
+    })
+  const inHomeTimezone = homeTimezone && homeTimezone === getUserTimezone()
+  const timezoneWarning = !inHomeTimezone && getTimezoneWarning(homeTimezone)
+
+  const setLocationFromStop = (locationType: 'from' | 'to') => {
+    const location = {
+      lat: stopData.lat,
+      lon: stopData.lon,
+      name: stopData.name
+    }
+    setLocation({ location, locationType, reverseGeocode: false })
+  }
+
+  const zoomToStop = () => {
+    zoomToPlace(map.default, stopData)
+  }
+
+  const onMouseEnter = () => {
+    zoomToStop()
+    setHoveredStop(stopData.gtfsId)
+  }
+
+  const onMouseLeave = () => {
+    setHoveredStop(undefined)
+  }
+
+  const operator = agencies.size
+    ? transitOperators?.find((o) => o.agencyId === Array.from(agencies)[0])
```

Comment:
In the case of shared stops, does it make any sense to show the first operator found in the list? An alternative is to show all operator logos, but maybe do it after the text about the stop ID.

Reply:
I think the intent was to check the size equals 1 rather than is greater than zero. That way for shared stops it'll display the generic stop icon.

Reply:
As written, this (or something else?) is showing the logo of the first operator for "shared" stops, i.e. stops with the same `stop_code`. The first operator changes depending on the order of stops from the nearby query, so the result is not consistent visually. Try for instance around Peachtree+10th in Atlanta.

Reply:
Oh yes, I was saying that the behavior you are seeing was a bug. The condition here was supposed to be `agencies.size === 1` but it wasn't. I changed it to that so that the behavior is as intended. Let me know if that works. I think this is fine vs listing out all the agency's icons, but let me know!